packages/alfa-dom/src/node/attribute.ts

```diff
@@ -93,6 +95,26 @@ export class Attribute extends Node {
     return path;
   }
 
+  /**
+   * @see https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#space-separated-tokens
+   */
+  public tokens(separator: string | RegExp = /\s+/): Array<string> {
+    return this._value.trim().split(separator).filter(not(isEmpty));
+  }
+
+  /**
+   * @see https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#enumerated-attribute
+   */
+  public enumerate(): Option<string>;
+
+  public enumerate<VALID extends string = string>(valid: VALID, ...rest: Array<VALID>): Option<VALID>;
```

Comment:
```suggestion
  /**
   * @see https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#enumerated-attribute
   */
  public enumerate<T extends string>(valid: T, ...rest: Array<T>): Option<T>;
```

A type variable is always declared using a single upper-case letter, with `T` being preferred as the type variable of highest importance, such as `Result<T, E>` and `Option<T>`.

The docblock should be included in both overloads to ensure that it always shows up in intellisense and not just when the first overload is used. The reason for making an overload instead of just relying on the implementation declaration is that the two overloads warrant separate documentation as they don't behave the same.

Reply:
Note that I do hate the "single letter possibly T" convention for cases where the type variable has some meaning (like here) (it does make full sense in `Option<T>` where it has no meaning).

I mean, if we compile litterature, we probably have thousands of pages on why and how we should give good names to variables. I'm having a hard time seeing this conclude by "oh yeah, if it's a type variable, just name it T"… That sounds very similar to advising to name every (code) variable i, j, k, …

Reply:
I absolutely agree that variables in general should have proper names. The way I see it though, type variables don't fall in the category of general variables; there should never be more than at most a couple of them in any given context, unlike variables in general. If you need to name them anything beyond a single letter in order to distinguish them then there's too many.

If a certain type variable has "special" meaning then we simply pick a letter that we find more appropriate, such as `Parser<I, T, E>` which uses `I` for `Input` and `E` for `Error`.